### DISCUSSION TITLE
Different format/style for footnote references instead of superscript

### DISCUSSION THREAD

**Discussion by chrarnold** [typst/typst#4448] [General]
### Description

I have an admittedly rather special use case that I still hope can somehow be tackled with typst, I don't want to switch to regular Latex because I am quite happy so far with typst.

In essence, I need a different style for how footnote references appear. It is NOT about the footnote line at the bottom of the page but rather the superscript reference to it. 

In my case, I must either not have additional horizontal space because of a footnote reference OR the reference (the number in superscript) must be aligned horizontally in a way that the subsequent character is perfectly aligned (I am using a monospace font). That is, a (partial) space is added after the footnote reference, so that the next character visually is aligned correctly. Can one choose a monospace font for footnotes? I find that depending on the number, alignment works or doesn't.

Example:
`TACTTTA`
`TA1CTT`

Here,  both texts should perfectly aligned vertically, irrespective of the exact footnote number (1, 5 10, 100, ...).

Alternatively, is it possible to avoid the superscript altogether and replace it with a different style, say underscore or a special background color or alike?

Couldn't find anything in the help section

Thank you for this amazing project!

### Use Case

Useful for cases when one needs each line to have exactly the same width / number of characters in the potential presence of footnotes.

**Comment by Mc-Zen** [comment]
Hi @chrarnold are you still looking for an answer to this question?

**Comment by Mc-Zen** [comment]
You can change the appearance of the footnote mark through a `show` rule on `footnote`
```typ

#show footnote: it => {
  let num = numbering(it.numbering, ..counter(footnote).at(here()))
  highlight(num, fill: red)
}
```
The footnote element gives us all we need to reconstruct the number. We just call `numbering()` with the current numbering pattern for footnotes (stored in `it`) and pass the current counter state for footnotes. Now we can do with the number whatever we like, e.g., show it underlined, highlighted etc. 

You can even keep the superscript style and place it in a box with the same width as all other characters in order to prevent messing up to spacing. 
```typ
#show footnote: it => {
  let num = numbering(it.numbering, ..counter(footnote).at(here()))
  box(width: measure[1].width, super(num))
}
```
![image](https://github.com/typst/typst/assets/52877387/ccadba74-de0f-423c-aae1-5b51c64f0f33)

This will of course fail a bit if the footnote number goes to two digits ^^

**Reply by Mc-Zen** [reply]
Although I'm actually having a hard time finding a monospaced font where the superscript numbers don't match the width of the other characters. If you use 
```typ
#set text(font: "my mono font")
```
in your document and **not**
```typ
#set super(typographic: false)
```
it produces monospaced output for superscript numbers out-of-the-box. But your specific font might not have typographic superscript numbers ...

**Comment by chrarnold** [comment]
Thanks a lot for the reply, it is appreciated!
I guess there is no easy way to put the footnote number on top of the character and not as superscript to the right, just to avoid any additional vertical space in the first place... I guess it is possible to completely omit the footnote number while still showing the footnote text in the footer, but then it's hard to understand which footnote text belongs where...

Or I do footnotes normally with a mono space font with proper superscript numbers and then I add a custom script to go through the raw file, and add "custom" line breaks to perfectly match the line width (which then also deals with the case when footprint numbers are two or three digits)

**Reply by Mc-Zen** [reply]
You're welcome!

> I guess there is no easy way to put the footnote number on top of the character and not as superscript to the right, just to avoid any additional vertical space in the first place

Actually, this is pretty easy!

```typ

  #show footnote: it => {
    let num = numbering(it.numbering, ..counter(footnote).at(here()))
    box(place(dy: -10pt, dx: -2pt, super(num))) 
  }
  ABGY
  
  ABF#footnote[asdasd]F
```
Just use `place` and put it in a box with `width` 0. 

![image](https://github.com/typst/typst/assets/52877387/41740a79-976f-4b95-bcdd-67fc7b571028)
